app/models/door_code.rb

```diff
@@ -0,0 +1,22 @@
+class DoorCode < ApplicationRecord
+  belongs_to :user
+
+  validates :code, presence: true
+  validates_uniqueness_of :code, case_sensitive: false
+end
+
+# == Schema Information
+#
+# Table name: door_codes
+#
+#  id         :integer          not null, primary key
+#  code       :string           not null
+#  enabled    :boolean          default(FALSE), not null
+#  created_at :datetime
+#  updated_at :datetime
+#  user_id    :integer
```

Comment:
I think `user_id` this should be a not null foreign key. It isn't showing that way in this autogenerated comment, which makes me worry that it is not.